### PR TITLE
api.git.clone

### PR DESCRIPTION
**Pull request by BaderBC**

### CODE REVIEW

**Review by maxidragon — APPROVED** [review]

File: internal/lua_tests/git_test.lua [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        return false
+    end
+
+    for _, file in files() do
+        if string.find(file.Path, "README.md") then
+            foundReadme = true
+        end
+    end
+
+    local _, err = api.sh.command("rm -rf " .. destPath)
```
Comment (cooligus) [inline internal/lua_tests/git_test.lua:25]: a bit dangerous (anyway it's deleted in go, isn't it?)

File: internal/lua_tests/lua_api_test.go [hunk 2]
```diff
@@ -91,14 +96,19 @@ func TestLuaApi(t *testing.T) {
...
 			err = script.afterTest()
 			if err != nil {
-				t.Fatalf("error occured during finalization stage of test '%s': %s", script.file, err)
+				logAndFail(t, "error occured during finalization stage of test '%s': %s", script.file, err)
 			}
 		}
 	}
 }
+
+func logAndFail(t *testing.T, format string, args ...interface{}) {
```
Comment (cooligus) [inline internal/lua_tests/lua_api_test.go:111]: What this function is supposed to be? Why not `t.Fatalf`?

File: pkg/api/git.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	}
+
+	if err := moveRecursively(g.FsVrct, tmpDir, destinationPath); err != nil {
+		return err
+	}
+
+	return os.RemoveAll(tmpDir)
+}
+
+func moveRecursively(fsVrct *vrctFs.VRCTFs, from, to string) error {
```
Comment (cooligus) [inline pkg/api/git.go:36]: I would move this function to vrct

**Review by cooligus — CHANGES_REQUESTED** [review]

File: internal/lua_tests/lua_api_test.go (hunk 2 above)
Comment (BaderBC) [reply internal/lua_tests/lua_api_test.go:111]: `t.Fatal` immediately ends execution, like `os.Exit(1)` 

logAndFail, logs and marks test as failed, but without ending execution

It allows test to make a cleanup

File: pkg/api/git.go (hunk 3 above)
Comment (BaderBC) [reply pkg/api/git.go:36]: 1) it is temporary solution, there should be other way in vrct to move files
2) it just uses vrct, there's no part of that code which is specific for vrct

File: internal/lua_tests/git_test.lua (hunk 1 above)
Comment (BaderBC) [reply internal/lua_tests/git_test.lua:25]: I isn't deleted in go, I will do that

Comment (BaderBC) [reply internal/lua_tests/git_test.lua:25]: Never mind

File: pkg/api/git.go (hunk 3 above)
Comment (cooligus) [reply pkg/api/git.go:36]: It is specific for VRCT.. It's kind of wrapper or helper around VRCT that can be outside `git.go`

**Review by cooligus — APPROVED** [review]

**Pull request merged**